api/cleanup_unused_api_indexes.py

```diff
@@ -0,0 +1,189 @@
+        base_url=credentials["endpoint"],
+        auth=(credentials["username"], credentials["password"]),
+    )
+
+
+def list_indexes(es_client):
+    """
+    Returns a list of indexes in the Elasticsearch cluster, sorted by index name.
+    """
+    resp = es_client.get("/_cat/indices", params={"format": "json"})
+    resp.raise_for_status()
+
+    return sorted(
+        [
+            {
+                "name": r["index"],
+                "size": r["store.size"],
+                "doc_count": int(r["docs.count"]),
+            }
+            for r in resp.json()
+        ],
+        key=lambda r: r["name"],
+    )
+
+
+def get_index_name(api_url):
+    """
+    Return the name of the index that this instance of the API is reading from.
+    """
+    search_templates_resp = httpx.get(
+        f"https://{api_url}/catalogue/v2/search-templates.json"
+    )
+    return search_templates_resp.json()["templates"][0]["index"]
+
+
+def maybe_cleanup_index(es_client, *, idx, prod_index_name, stage_index_name):
+    """
+    Decide if we're going to clean up this index, and if so, double-check with the user.
+    """
+    if not idx["name"].startswith(("works-", "images-")):
+        return
+
+    prod_works_index_name = prod_index_name
+    prod_images_index_name = prod_index_name.replace("works-", "images-")
+
+    stage_works_index_name = stage_index_name
+    stage_images_index_name = stage_index_name.replace("works-", "images-")
+
+    click.echo(
+        f"\nConsidering %s (%s docs, %s)"
+        % (
+            click.style(idx["name"], "blue"),
+            humanize.intcomma(idx["doc_count"]),
+            idx["size"],
+        )
+    )
+
+    # We should never delete the prod or staging indexes -- this would cause an instant
+    # outage in the API, which is very bad.
+    if idx["name"] == prod_works_index_name or idx["name"] == prod_images_index_name:
+        click.echo(
+            f"This index will {click.style('not be deleted', 'green')} -- it is the prod API"
+        )
+        return
+
+    if idx["name"] == stage_works_index_name or idx["name"] == stage_images_index_name:
+        click.echo(
+            f"This index will {click.style('not be deleted', 'green')} -- it is the stage API"
+        )
+        return
+
+    # We consider deleting an index if it sorts lexicographically lower than both the
+    # prod and staging APIs.  e.g. works-2001-01-01 < works-2002-02-02
+    #
+    # We skip indexes that sort higher, because this might be an index that we're currently
+    # reindexing into, but we haven't pointed an API at yet.  In general, indexes go forward,
+    # not backward.
+    #
+    # We still ask the user to confirm they really want to delete this index, just in case.
+    # We should never offer to delete an index that would cause an outage, but they might
+    # skip an index if, say, they've only just promoted a new index to prod and they
+    # want to be able to roll back to this index.
```

Comment:
I almost wonder if this is a requirement, keeping a rollbackable index? I'm just not sure how we would work that out...

Saying that, maybe worth a post-using and testing this script.